src/object.rs

```diff
@@ -0,0 +1,122 @@
+use std::sync::mpsc;
+
+use mint;
+
+use hub::{Message, Operation, SubNode};
+use node::{NodeInfo, NodePointer};
+use scene::Scene;
+
+//Note: no local state should be here, only remote links
+/// `Object` represents an entity that can be added to the scene.
+///
+/// There is no need to use `Object` directly, there are specific wrapper types
+/// for each case (e.g. [`Camera`](struct.Camera.html),
+/// [`AmbientLight`](struct.AmbientLight.html),
+/// [`Mesh`](struct.Mesh.html), ...).
+#[derive(Clone, Debug)]
+pub struct Object {
+    pub(crate) node: NodePointer,
+    pub(crate) tx: mpsc::Sender<Message>,
+}
+
+impl AsRef<NodePointer> for Object {
+    fn as_ref(&self) -> &NodePointer {
+        &self.node
+    }
+}
+
+impl Object {
+    /// Invisible objects are not rendered by cameras.
+    pub fn set_visible(&mut self, visible: bool) {
+        let msg = Operation::SetVisible(visible);
+        let _ = self.tx.send((self.node.downgrade(), msg));
+    }
+
+    /// Rotates object in the specific direction of `target`.
+    pub fn look_at<P>(&mut self, eye: P, target: P, up: Option<mint::Vector3<f32>>)
```

Comment:
`eye` and `target` should have different type parameters. That can be fixed in another time though.